.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,122 @@
+name: Run Tests
+
+on:
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+env:
+  DREMIO_SOFTWARE_HOST: ${{ secrets.SOFTWARE_HOST }}
+  DREMIO_CLOUD_HOST: ${{ secrets.CLOUD_HOST }}
+  DREMIO_PORT: ${{ secrets.DREMIO_PORT }}
+  DREMIO_SOFTWARE_USER: ${{ secrets.DREMIO_SOFTWARE_USER }}
+  DREMIO_PW: ${{ secrets.DREMIO_SOFTWARE_PASSWORD }}
+  DREMIO_SOFTWARE_PAT: ${{ secrets.DREMIO_SOFTWARE_PAT }}
+  DREMIO_CLOUD_PAT: ${{ secrets.DREMIO_CLOUD_PAT }}
+  DREMIO_CLOUD_USER: ${{ secrets.DREMIO_CLOUD_USER }}
+  DREMIO_CLOUD_PROJECT: ${{ secrets.DREMIO_CLOUD_PROJECT }}
+  DREMIO_SOURCE: ${{ secrets.DREMIO_SOURCE}}
+  DREMIO_SPACE: ${{ secrets.DREMIO_SPACE}}
+  DBT_TEST_USER_1: ${{ secrets.DBT_TEST_USER_1}}
+  DBT_TEST_USER_2: ${{ secrets.DBT_TEST_USER_2}}
+  DBT_TEST_USER_3: ${{ secrets.DBT_TEST_USER_3}}
+
+jobs:
+  run-functional-tests_cloud:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout the project
+        uses: actions/checkout@v2
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.10.0
+      - name: Install requirements
+        run: pip install -r dev_requirements.txt
+      - name: Set env variables
+        run: |
+          echo "DREMIO_CLOUD_HOST=$DREMIO_CLOUD_HOST" >> $GITHUB_ENV
```

Comment:
This essentially sets takes each of our secrets and writes them to the GITHUB_ENV environment file. From there it can be loaded into our conftest.py file.